parts/k8s/cloud-init/artifacts/kms-keyvault-key.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+set +x
+set -euo pipefail
+
+AZURE_JSON_PATH="/etc/kubernetes/azure.json"
+SERVICE_PRINCIPAL_CLIENT_ID=$(jq -r '.aadClientId' ${AZURE_JSON_PATH})
+SERVICE_PRINCIPAL_CLIENT_SECRET=$(jq -r '.aadClientSecret' ${AZURE_JSON_PATH})
+TENANT_ID=$(jq -r '.tenantId' ${AZURE_JSON_PATH})
+KMS_KEYVAULT_NAME=$(jq -r '.providerVaultName' ${AZURE_JSON_PATH})
+KMS_KEY_NAME=$(jq -r '.providerKeyName' ${AZURE_JSON_PATH})
```

Comment:
where do you set the value name and the key name in azure.json?

Reply:
These 2 values are already set when azure.json is created the first time - https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/artifacts/cse_config.sh#L212-L215

Reply:
I was referring to how the keyname is passed into the cse_config.sh. But I see you are hardcoding it in the ARM template:
`[concat(variables('clusterKeyVaultName'), '/', 'k8s')]",`